week1/project/EcommerceVika/ecommerce-shop-project/src/App.js

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from "react";
+import "./App.css";
+import Categories from "./components/Categories";
+import Products from "./components/Products";
+import FilteredProductsByCategory from "./components/FilteredProductsByCategory";
+import allCategories from "./fake-data/all-categories.js";
+import allProducts from "./fake-data/all-products.js";
+
+function App() {
+  const [selectedCategory, setSelectedCategory] = useState("");
+
+  const handleCategoryClick = (category) => {
+    console.log(`Clicked category: ${category}`);
+    setSelectedCategory(category);
+  };
+  
+  const filteredProducts = FilteredProductsByCategory(allProducts, selectedCategory);
+
+  return (
+    <div className="App">
+      <Categories
+        categories={allCategories}
+        selectedCategory={selectedCategory}
+        handleCategoryClick={handleCategoryClick}
+        />
+        <Products 
+        filteredProducts={filteredProducts}
+        />
```

Comment:
indentation is very important .. at first I thought that `Products` is a child of `Categories` I do recommend you use an extension like [Pretier ](https://prettier.io/)of [ESlint ](https://eslint.org/) or if you prefer doing it manually but you need to take care about indentations

Reply:
I will read about indentation. It's how lines go after each other? Anna told me about this problem... but I'm so tired, when it works, that I usually forget((( I'll try to remember. Thank you for advise